Sources/SpeziViews/Model/ViewState.swift

```diff
@@ -10,6 +10,52 @@ import Foundation
 
 
 /// Allows SwiftUI views to keep track of their state and communicate to outside views.
+///
+/// A `ViewState` provides a built-in mechanism for tracking the state of a Spezi UI component.
+/// A view can be in either an `idle`, `processing`, or `error` `ViewState`.
+///
+/// ```swift
+/// import SpeziViews
+/// import SwiftUI
+///
+/// enum ProcessingError: LocalizedError {
+///     case resourceError
+/// }
+///
+/// func doSomeProcessing() throws {
+///     // Function may throw an error
+///     throw ProcessingError.resourceError
+/// }
+///
+/// struct OperationStateTestView: View {
+///     @State private var viewState: ViewState = .idle
+///
+///     var body: some View {
+///         VStack {
+///             Button("Action") {
+///                 viewState = .processing
+///                 do {
+///                     try doSomeProcessing()
+///                     viewState = .idle
+///                 } catch let error as LocalizedError {
+///                     viewState = .error(error)
+///                 } catch {
+///                     print("An unknown error occurred: \(error)")
+///                 }
+///             }
+///             .viewStateAlert(state: $viewState)
+///         }
+///     }
+/// }
+/// ```
+///
+/// The ``SwiftUI/View/viewStateAlert(state:)-4wzs4`` view modifier can be used to automatically notify users when the
+/// `ViewState` enters an error state.
+///
+/// > Tip:
+/// > To avoid having to handle state changes manually (as seen above), a `ViewState` can be mapped to an
+/// > `OperationState`, allowing the `ViewState` to change automatically when the application's internal state changes.
```

Comment:
I am not sure if this really expresses what `OperationState` does. Operation state rather allows you to define more custom cases that would be relevant to your application and then maps it back to `ViewState` to make it easy to use the same `viewStateAlert` modifiers and other elements. We might want to rephrase this a bit.